tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_Constraints.cpp

```diff
@@ -726,3 +742,38 @@ SPECTRE_TEST_CASE("Unit.Evolution.Systems.GeneralizedHarmonic.FConstraint",
   test_f_constraint_random<3, Frame::Inertial, double>(
       std::numeric_limits<double>::signaling_NaN());
 }
+
+SPECTRE_TEST_CASE(
+    "Unit.Evolution.Systems.GeneralizedHarmonic.ConstraintEnergyNorm",
+    "[Unit][Evolution]") {
+  pypp::SetupLocalPythonEnvironment local_python_env{
+      "Evolution/Systems/GeneralizedHarmonic/"};
+
+  // Test the constraint energy normalization with random numbers
+  test_constraint_energy_normalization<1, Frame::Grid, DataVector>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_constraint_energy_normalization<1, Frame::Inertial, DataVector>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_constraint_energy_normalization<1, Frame::Grid, double>(
+      std::numeric_limits<double>::signaling_NaN());
+  test_constraint_energy_normalization<1, Frame::Inertial, double>(
+      std::numeric_limits<double>::signaling_NaN());
+
+  test_constraint_energy_normalization<2, Frame::Grid, DataVector>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_constraint_energy_normalization<2, Frame::Inertial, DataVector>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_constraint_energy_normalization<2, Frame::Grid, double>(
+      std::numeric_limits<double>::signaling_NaN());
+  test_constraint_energy_normalization<2, Frame::Inertial, double>(
+      std::numeric_limits<double>::signaling_NaN());
+
+  test_constraint_energy_normalization<3, Frame::Grid, DataVector>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_constraint_energy_normalization<3, Frame::Inertial, DataVector>(
+      DataVector(4, std::numeric_limits<double>::signaling_NaN()));
+  test_constraint_energy_normalization<3, Frame::Grid, double>(
+      std::numeric_limits<double>::signaling_NaN());
+  test_constraint_energy_normalization<3, Frame::Inertial, double>(
+      std::numeric_limits<double>::signaling_NaN());
+}
```

Comment:
add an analytic test

Reply:
I didn’t do an analytic test because I don’t have an analytic formula for this, and deriving one would be tedious. Do you know of an analytic solution I can use? If not, I’d really prefer just using python tests on this one, but failing that, can it just be Schwarzschild or some made up simple function for each derivative?

Reply:
Schwarzschild would be fine